### PR TITLE
Add VCTK dataset

### LINKED ISSUE #1837: Add VCTK
## Adding a Dataset
- **Name:** *VCTK*
- **Description:** *This CSTR VCTK Corpus includes speech data uttered by 110 English speakers with various accents. Each speaker reads out about 400 sentences, which were selected from a newspaper, the rainbow passage and an elicitation paragraph used for the speech accent archive.*
- **Paper:** Homepage: https://datashare.ed.ac.uk/handle/10283/3443
- **Data:** https://datashare.ed.ac.uk/handle/10283/3443
- **Motivation:** Important speech dataset
- **TFDatasets Implementation**: https://www.tensorflow.org/datasets/catalog/vctk

If interested in tackling this issue, feel free to tag @patrickvonplaten

Instructions to add a new dataset can be found [here](https://github.com/huggingface/datasets/blob/master/ADD_NEW_DATASET.md).

### PR DESCRIPTION
**Pull request by jaketae**
Fixes #1837.

### CODE REVIEW

**Comment by jaketae** [comment]
Hello @patrickvonplaten, I hope it's okay to ping you with a (dumb) question. I've been trying to get `dl_manager.download_and_extract(_DL_URL)` to work with no avail. I verified that this is a problem on two different machines (lab server, GCP), so I doubt it's an issue with network connectivity. Here is the full trace.

```
(venv) (base) jaketae@jake-gpu1:~/documents/datasets$ datasets-cli test datasets/vctk --save_infos --all_configs
Testing builder 'main' (1/1)
Downloading and preparing dataset vctk/main to /home/jaketae/.cache/huggingface/datasets/vctk/main/0.9.2/2bfa52a93469fa9d6d4b1831c6511db5442b9f4e48620aef2bc3890d7a5268a8...
Traceback (most recent call last):
  File "/home/jaketae/documents/datasets/venv/bin/datasets-cli", line 33, in <module>
    sys.exit(load_entry_point('datasets', 'console_scripts', 'datasets-cli')())
  File "/home/jaketae/documents/datasets/src/datasets/commands/datasets_cli.py", line 33, in main
    service.run()
  File "/home/jaketae/documents/datasets/src/datasets/commands/test.py", line 146, in run
    builder.download_and_prepare(
  File "/home/jaketae/documents/datasets/src/datasets/builder.py", line 593, in download_and_prepare
    self._download_and_prepare(
  File "/home/jaketae/documents/datasets/src/datasets/builder.py", line 659, in _download_and_prepare
    split_generators = self._split_generators(dl_manager, **split_generators_kwargs)
  File "/home/jaketae/.cache/huggingface/modules/datasets_modules/datasets/vctk/2bfa52a93469fa9d6d4b1831c6511db5442b9f4e48620aef2bc3890d7a5268a8/vctk.py", line 76, in _split_generators
    root_path = dl_manager.download_and_extract(_DL_URL)
  File "/home/jaketae/documents/datasets/src/datasets/utils/download_manager.py", line 283, in download_and_extract
    return self.extract(self.download(url_or_urls))
  File "/home/jaketae/documents/datasets/src/datasets/utils/download_manager.py", line 195, in download
    downloaded_path_or_paths = map_nested(
  File "/home/jaketae/documents/datasets/src/datasets/utils/py_utils.py", line 234, in map_nested
    return function(data_struct)
  File "/home/jaketae/documents/datasets/src/datasets/utils/download_manager.py", line 216, in _download
    return cached_path(url_or_filename, download_config=download_config)
  File "/home/jaketae/documents/datasets/src/datasets/utils/file_utils.py", line 298, in cached_path
    output_path = get_from_cache(
  File "/home/jaketae/documents/datasets/src/datasets/utils/file_utils.py", line 608, in get_from_cache
    raise ConnectionError(f"Couldn't reach {url}")
ConnectionError: Couldn't reach https://datashare.is.ed.ac.uk/bitstream/handle/10283/3443/VCTK-Corpus-0.92.zip
```

On my local, however, the URL correctly points to the download zip file. My admittedly naive guess is that the website is web-crawler or scraper proof (requiring specific headers, etc.), but I also think I might have just missed a very basic step in the process.

Apologies for the delayed PR, and TIA for the help!

**Comment by patrickvonplaten** [comment]
Hey @jaketae, 

Hmm, yeah I don't know really either - the link also works correctly for me when doing:

```
wget https://datashare.is.ed.ac.uk/bitstream/handle/10283/3443/VCTK-Corpus-0.92.zip
```

I think however that I had a similar problem previously with Edinburgh's (`.ed.ac.uk`) websites which I solved with the following hack. Not sure if this could be the same problem here...
https://github.com/huggingface/datasets/blob/e1104ad5d3e83f8b1571e0d6fef4fdabf0a1fde5/datasets/ami/ami.py#L364

**Comment by patrickvonplaten** [comment]
The AMI dataset is stored under a different website though it seems: `"https://groups.inf.ed.ac.uk/ami/AMICorpusMirror//amicorpus/{}/audio/{}"`

so not 100p sure if this solves the problem

**Comment by jaketae** [comment]
Hi @patrickvonplaten,

Thanks for the feedback! Sadly, disabling multi-processing didn't cut it for me. 

I've been looking at VCTK code in [`torchaudio`](https://pytorch.org/audio/stable/_modules/torchaudio/datasets/vctk.html) and [`tfds`](https://github.com/tensorflow/datasets/blob/master/tensorflow_datasets/audio/vctk.py). I don't think they're using a hack to accomplish this, so I'll try to look into it to see if I can pinpoint the cause. I'll keep you in the loop here. Thank you!

**Comment by jaketae** [comment]
Hi @patrickvonplaten, 

After more investigation, I found that simply increasing `etag_timeout` in `get_from_cache` from 10 to 100 solved it. However, unless I'm missing something, an issue is that `etag_timeout` is basically hard-coded as a default parameter because `cached_path`, which calls `get_from_cache` has no way of modifying the default. I can think of two solutions.

* Simply increase the default to 100
* Allow `etag_timeout` to be modifiable on a per-dataset basis by integrating it to `download_kwargs` or some other object

Thank you!

**Comment by patrickvonplaten** [comment]
I think in this case we can increase the `etag_timeout` - what do you think @lhoestq @albertvillanova ?

**Comment by lhoestq** [comment]
Yes let's increase it to 100 for the moment. Later we can see if it really needed to move it into `download_config` or not

**Comment by jaketae** [comment]
Thanks for the feedback @patrickvonplaten @lhoestq, I'll continue working on this in that direction!

**Comment by jaketae** [comment]
Hello @patrickvonplaten, VCTK is ready for review! 

```python
>>> from datasets import load_dataset
>>> ds = load_dataset("vctk")
Using the latest cached version of the module from /home/lily/jt856/.cache/huggingface/modules/datasets_modules/datasets/vctk/b7aa278182de3a7aa2897cbd12c1e19f1af9840a2ead69a6d710fdbc1d2df02a (last modified on Sat Dec 25 00:47:31 2021) since it couldn't be found locally at vctk., or remotely on the Hugging Face Hub.
Reusing dataset vctk (/home/lily/jt856/.cache/huggingface/datasets/vctk/main/0.9.2/b7aa278182de3a7aa2897cbd12c1e19f1af9840a2ead69a6d710fdbc1d2df02a)
100%|████████████████████████████████████████████████████████████████████████████████| 1/1 [00:00<00:00, 198.35it/s]
>>> len(ds["train"])
88156
>>> ds["train"][0]
{'speaker_id': 'p225', 'audio': {'path': '/home/lily/jt856/.cache/huggingface/datasets/downloads/extracted/8ed7dad05dfffdb552a3699777442af8e8ed11e656feb277f35bf9aea448f49e/wav48_silence_trimmed/p225/p225_001_mic1.flac', 'array': array([0.00485229, 0.00689697, 0.00619507, ..., 0.00811768, 0.00836182,
       0.00854492], dtype=float32), 'sampling_rate': 48000}, 'file': '/home/lily/jt856/.cache/huggingface/datasets/downloads/extracted/8ed7dad05dfffdb552a3699777442af8e8ed11e656feb277f35bf9aea448f49e/wav48_silence_trimmed/p225/p225_001_mic1.flac', 'text': 'Please call Stella.', 'text_id': '001', 'age': '23', 'gender': 'F', 'accent': 'English', 'region': 'Southern England', 'comment': ''}
```

Let me know if you see any rough edges. Also, belated merry Christmas and a happy new year!

**Review by patrickvonplaten — APPROVED** [review]
Great looks good to me!

File: datasets/vctk/README.md [hunk 1]
```diff
@@ -0,0 +1,154 @@
+---
+annotations_creators:
+- expert-generated
+language_creators:
+- crowdsourced
+languages:
+- en-US, en-GB
```
Comment (lhoestq) [inline datasets/vctk/README.md:7]:
```suggestion
- en
```

File: datasets/vctk/README.md [hunk 2]
```diff
@@ -0,0 +1,154 @@
...
+licenses:
+- cc-by-4.0
+multilinguality:
+- monolingual
+pretty_name: VCTK
+size_categories:
+- unknown
```
Comment (lhoestq) [inline datasets/vctk/README.md:14]:
```suggestion
- 10K<n<100K
```

File: datasets/vctk/README.md [hunk 3]
```diff
@@ -0,0 +1,154 @@
...
+source_datasets:
+- original
+task_categories:
+- automatic-speech-recognition
+task_ids:
+- speech-recognition
```
Comment (lhoestq) [inline datasets/vctk/README.md:20]:
```suggestion
- speech-recognition
paperswithcode_id: vctk
```

File: datasets/vctk/README.md [hunk 4]
```diff
@@ -0,0 +1,154 @@
...
+- `text_id`: Unique text ID
+- `age`: Speaker's age
+- `gender`: Speaker's gender
+- `accent`: Speaker's accent
+- `region`: Speaker's region, if annotation exists
+- `comment`: Miscellaneous comments, if any
+
+### Data Splits
+
+[More Information Needed]
```
Comment (lhoestq) [inline datasets/vctk/README.md:92]:
```suggestion
The dataset has no predefined splits.
```

File: datasets/vctk/README.md [hunk 5]
```diff
@@ -0,0 +1,154 @@
...
+
+### Licensing Information
+
+[More Information Needed]
+
+### Citation Information
+
+[More Information Needed]
```
Comment (lhoestq) [inline datasets/vctk/README.md:150]:
````suggestion
```bibtex
@inproceedings{Veaux2017CSTRVC,
    title        = {CSTR VCTK Corpus: English Multi-speaker Corpus for CSTR Voice Cloning Toolkit},
    author       = {Christophe Veaux and Junichi Yamagishi and Kirsten MacDonald},
    year         = 2017
}
```
````

File: datasets/vctk/README.md (hunk 3 above)
Comment (lhoestq) [inline datasets/vctk/README.md:20]:
```suggestion
task_categories:
- speech-processing
task_ids:
- automatic-speech-recognition
- ```

**Review by lhoestq — COMMENTED** [review]
Thanks ! It looks all good :)

Let me just fix the dataset card:

File: datasets/vctk/README.md [hunk 6]
```diff
@@ -0,0 +1,161 @@
...
+
+### Languages
+
+[More Information Needed]
+
+## Dataset Structure
+
+### Data Instances
+
+[More Information Needed]
```
Comment (lhoestq) [inline datasets/vctk/README.md:74]: Do you think you could provide an example from the dataset here ?

You can get one by using `load_dataset("./datasets/vctk")` and doing `dataset["train"][0]` for example.

Comment (jaketae) [reply datasets/vctk/README.md:74]: Thank you for brushing up the dataset card. Absolutely, I'll add an example shortly.

**Review by lhoestq — APPROVED** [review]
Thanks !

The CI failure is unrelated to this PR, we're good to merge :)

File: src/datasets/utils/file_utils.py [hunk 7]
```diff
@@ -499,7 +499,7 @@ def get_from_cache(
     cache_dir=None,
     force_download=False,
     proxies=None,
-    etag_timeout=10,
+    etag_timeout=100,
```
Comment (patrickvonplaten) [inline src/datasets/utils/file_utils.py:502]: @michaelbenayoun - increasing this might help in your case

**Pull request merged**